nbconvert/nbconvertapp.py

```diff
@@ -30,6 +31,36 @@
 #Classes and functions
 #-----------------------------------------------------------------------------
 
+class CaselessStrEnumOrDottedObjectName(CaselessStrEnum, DottedObjectName):
+    """Either an Enum whose value must be in a given sequence, or a Dotted Object Name
+
+    Exporter can be either alias of built-ins exporter that are white-listed
+    and are simple strings ('html', 'latex', 'pdf', ... ) Or a dotted object
+    name that will be imported at run-time and instantiate with the right
+    configuration.
+    """
+
+    def __init__(self, values, default_value=Undefined, **metadata):
+        self.values = values
+        Enum.__init__(self, values=values, default_value=default_value, **metadata)
+
+
+    def validate(self, obj, value):
+        if '.' not in value:
+            CaselessStrEnum.validate(self, obj, value)
+            pass
+        else: 
+            DottedObjectName.validate(self, obj, value)
```

Comment:
Should we be using Type instead of DottedObjectName? In most cases where we have DottedObjectName, Type is actually the right thing to use.